src/bootupd.rs

```diff
@@ -489,6 +490,149 @@ pub(crate) fn client_run_validate() -> Result<()> {
     Ok(())
 }
 
+pub(crate) fn client_run_migrate() -> Result<()> {
```

Comment:
I'd add `#[context("Migrating to static grub config")]` here

Reply:
(Which would then avoid the duplicate "Not performing migration" in all the error contexts)